domino_environments/_environments.py

```diff
@@ -325,8 +347,47 @@ def create_revision(
             headers={"Content-Type": "application/x-www-form-urlencoded"},
         )
 
+    #TODO: add some paging logic in here. search 10/20 at a time and fail if not found
+    def get_build_status(self, environment: Environment, revision_id: str = None):
+        """Gather current status of build job for a given environment/revision
+
+        Args:
+            environment: The environment object.
+            revision_id: The ID of the revision, defaults to the environment's active revision.
+        """
+        revision_id = revision_id or environment.active_revision.get("id")
+        summary = self._get_revision_summary(environment.id)
+        return summary['buildStatus'][revision_id]
+
+    def get_build_logs(self, environment: Environment, revision_id: str = None):
+        """Gather logs of build job for a given environment/revision
+
+        Args:
+            environment: The environment object.
+            revision_id: The ID of the revision, defaults to the environment's active revision.
+        """
+        revision_id = revision_id or environment.active_revision.get("id")
+        summary = self._get_revision_summary(environment.id)
+        logs_url = summary['buildLogsUrl'][revision_id]
+
+        url = self._routes.build_logs(logs_url)
+        res = self.request_manager.get(url)
+
+        regex = re.compile((r'<td class="line".*?>(.*)</td>'))
```

Comment:
Does this work, because I'm viewing HTML source code for the logs url and it's showing `<span>` instead of `<td>`.

Reply:
I tested on 4.6, from the API the actual log lines look like this. There is a <span> but the actual log line is separate:
```
<tr class="log-hit forge-build log" data-timeNano="1635259874823775998">
  <td><span class="timestamp" data-millis="1635259874823" data-format="MMM DD YYYY HH:mm:ss ZZ"></span></td>
  <td class="line" style="white-space: pre">Pushing image &quot;ubsaks.azurecr.io/dom46/buildneweragain/environment:6178151b88be3c08a6236fa4-1&quot;</td>
  <td>aks-compute-32127267-vmss00000a</td>
  <td>domino-build-6178151b88be3c08a6236fa8-v489d</td>
  <td>forge-build</td>
</tr>

<tr class="log-hit forge-build log" data-timeNano="1635259875869476061">
  <td><span class="timestamp" data-millis="1635259875869" data-format="MMM DD YYYY HH:mm:ss ZZ"></span></td>
  <td class="line" style="white-space: pre">Pushed image &quot;ubsaks.azurecr.io/dom46/buildneweragain/environment:6178151b88be3c08a6236fa4-1&quot;</td>
  <td>aks-compute-32127267-vmss00000a</td>
  <td>domino-build-6178151b88be3c08a6236fa8-v489d</td>
  <td>forge-build</td>
</tr>
```

Reply:
Just to add, i'm open to a better method of parsing these out. One option is BeautifulSoup, but i wanted to avoid adding more dependencies

Reply:
Ok sounds good, I think the browser rendered version is a bit different.
This works for now, we can add beautifulsoup in later if we start doing more of this stuff.